spark-jobs/src/main/scala/za/co/absa/enceladus/common/JobCmdConfig.scala

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.common
+
+import org.apache.spark.storage.StorageLevel
+import scopt.OParser
+import za.co.absa.enceladus.dao.auth.{InvalidMenasCredentialsFactory, MenasCredentialsFactory, MenasKerberosCredentialsFactory, MenasPlainCredentialsFactory}
+
+import scala.util.matching.Regex
+
+
+trait JobCmdConfig[R] {
+  def withDatasetName(value: String): R
+  def withDatasetVersion(value: Int): R
+  def withReportDate(value: String): R
+  def withReportVersion(value: Option[Int]): R
+  def withMenasCredentialsFactory(value: MenasCredentialsFactory): R
+  def withPerformanceMetricsFile(value: Option[String]): R
+  def withFolderPrefix(value: Option[String]): R
+  def withPersistStorageLevel(value: Option[StorageLevel]): R
+
+  def datasetName: String
+  def reportDate: String
+  def menasCredentialsFactory: MenasCredentialsFactory
+  def datasetVersion: Int
+  def reportVersion: Option[Int]
+  def performanceMetricsFile: Option[String]
+  def folderPrefix: Option[String]
+  def persistStorageLevel: Option[StorageLevel]
+}
+
+object JobCmdConfig {
+
+  def jobConfigParser[R <: JobCmdConfig[R]]: OParser[_, R] = {
```

Comment:
Mark this function to ignore scalastyle warning of being too long. It's legit to be like this.